File: ShoppingLiveData/app/src/main/java/dk/itu/shoppinglivedata/ListFragment.java
package dk.itu.shoppinglivedata;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

public class ListFragment extends Fragment {
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        final View v = inflater.inflate(R.layout.fragment_list, container, false);
        final TextView listThings = v.findViewById(R.id.listItems);
        final ShoppingViewModel viewModel = new ViewModelProvider(requireActivity()).get(ShoppingViewModel.class);
        viewModel.getUiState().observe(getViewLifecycleOwner(), uiState -> listThings.setText("Shopping List" + uiState.listItems));
        return v;
    }
}

File: ShoppingV4KotlinUnitTest/app/src/main/java/dk/itu/shoppingv4kotlinunittest/ListFragment.kt
package dk.itu.shoppingv4kotlinunittest

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import java.util.Observable
import java.util.Observer

class ListFragment : Fragment(), Observer {
    private lateinit var itemsDB: ItemsDB
    private lateinit var listThings: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        itemsDB = ItemsDB.get()
        itemsDB.addObserver(this)
    }

    override fun update(observable: Observable?, data: Any?) {
        listThings.text = "Shopping List" + itemsDB.listItems()
    }
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val v = inflater.inflate(R.layout.fragment_list, container, false)
        listThings = v.findViewById(R.id.listItems)
        listThings.text = "Shopping List" + itemsDB.listItems()
        return v
    }
}